services/hanke-service/src/test/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusServiceTest.kt

```diff
@@ -0,0 +1,139 @@
+package fi.hel.haitaton.hanke.hakemus
+
+import assertk.Assert
+import assertk.all
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import assertk.assertions.isNotNull
+import assertk.assertions.prop
+import assertk.assertions.single
+import fi.hel.haitaton.hanke.HankeEntity
+import fi.hel.haitaton.hanke.application.ApplicationContactType
+import fi.hel.haitaton.hanke.application.ApplicationData
+import fi.hel.haitaton.hanke.application.ApplicationNotFoundException
+import fi.hel.haitaton.hanke.application.ApplicationRepository
+import fi.hel.haitaton.hanke.application.CableReportApplicationData
+import fi.hel.haitaton.hanke.asJsonResource
+import fi.hel.haitaton.hanke.factory.ApplicationFactory
+import fi.hel.haitaton.hanke.factory.HakemusyhteystietoFactory
+import fi.hel.haitaton.hanke.factory.HakemusyhteystietoFactory.withYhteyshenkilo
+import fi.hel.haitaton.hanke.factory.HankeFactory
+import fi.hel.haitaton.hanke.factory.HankeKayttajaFactory
+import io.mockk.checkUnnecessaryStub
+import io.mockk.clearAllMocks
+import io.mockk.confirmVerified
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import io.mockk.verifySequence
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+
+private const val USERNAME = "test"
+
+class HakemusServiceTest {
+    private val applicationRepository: ApplicationRepository = mockk()
+
+    private val hakemusService: HakemusService = HakemusService(applicationRepository)
+
+    companion object {
+        private val applicationData: CableReportApplicationData =
+            "/fi/hel/haitaton/hanke/application/applicationData.json".asJsonResource()
+    }
+
+    @BeforeEach
+    fun cleanup() {
+        clearAllMocks()
+    }
+
+    @AfterEach
+    fun verifyMocks() {
+        checkUnnecessaryStub()
+        confirmVerified(
+            applicationRepository,
+        )
+    }
+
+    @Nested
+    inner class HakemusResponse {
```

Comment:
These don't seem to test anything the integration tests wouldn't test.

Reply:
Deleted the test.